src/utils/getResponse.ts

```diff
@@ -47,7 +51,14 @@ export const getResponse = async <
       return executionResult
     }
 
-    const result = await handler.run(request, resolutionContext)
+    const [error, result] = await until(() =>
+      handler.run(request, resolutionContext),
```

Comment:
I do like checking for exceptions immediately around `handler.run` but I don't think there are many things that may throw in `getResponse` on our part. I don't think that making this function accept an `emitter` is a good idea, it should be a function of handlers to mocked response, with as little side-effects as possible. 

I'd probably try moving this logic around `getResponse` and gather feedback from developers. We are unlikely to experience exceptions out of handler's context, but if we do we can always revisit—the surface at which we check for exception is an implementation detail, so we can change it at any time without introducing a breaking change.

Reply:
How pass the responsible handler back to `getResponse` in an elegant way?

I still need to wrap `hander.run`
```ts
    const [error, result] = await until(() =>
      handler.run(request, resolutionContext),
    )

    if (error) {
      // Don't emit anymore (emitter is not passed down to keep function pure)
      // emitter.emit('request:unhandledException', error, request, handler)
   
      // Somehow pass the handler
      const error = new Error()
      // Not great
      error.handler = handler // Property 'handler' does not exist on type 'Error'.
      throw error
    }
```